docs/source/conceptual_guides/oft.md

```diff
@@ -0,0 +1,105 @@
+    <img src="https://github.com/wy1iu/butterfly-oft/blob/main/assets/BOFT_comparison.png"/>
+</div>
+
+
+Serving a different alterative to LoRA, BOFT carries many advantages: 
+
+* BOFT proposes a simple yet generic way to finetune pretrained models to downstream tasks, yielding a better preservation of pretraining knowledge and a better parameter efficiency.
+* Through the orthogonality, BOFT introduces a structural constraint, i.e., keeping the [hyperspherical energy](https://arxiv.org/abs/1805.09298) unchanged during finetuning. This can effectively reduce the forgetting of pretraining knowledge.
+* BOFT uses the butterfly factorization to efficiently parameterize the orthogonal matrix, which yields a compact yet expressive learning space (i.e., hypothesis class).
+* The sparse matrix decomposition in BOFT brings in additional inductive biases that are beneficial to generalization.
+
+In principle, BOFT can be applied to any subset of weight matrices in a neural network to reduce the number of trainable parameters. Given the target layers for injecting BOFT parameters, the number of trainable parameters can be determined based on the size of the weight matrices.
+
+## Merge OFT/BOFT weights into the base model
+
+Similar to LoRA, the weights learned by OFT/BOFT can be integrated into the pretrained weight matrices using the merge_and_unload() function. This function merges the adapter weights with the base model which allows you to effectively use the newly merged model as a standalone model.
+
+<div class="flex justify-center">
+    <img src="https://github.com/wy1iu/butterfly-oft/blob/main/assets/boft_merge.png"/>
+</div>
+
+This works because during training, the orthogonal weight matrix (R in the diagram above) and the pretrained weight matrices are separate. But once training is complete, these weights can actually be merged (multiplied) into a new weight matrix that is equivalent.
+
+## Utils for OFT / BOFT
+
+### Common OFT / BOFT parameters in PEFT
+
+As with other methods supported by PEFT, to fine-tune a model using OFT or BOFT, you need to:
+
+1. Instantiate a base model.
+2. Create a configuration (`OFTConfig` or `BOFTConfig`) where you define OFT/BOFT-specific parameters.
+3. Wrap the base model with `get_peft_model()` to get a trainable `PeftModel`.
+4. Train the `PeftModel` as you normally would train the base model.
+
+
+### BOFT-specific paramters
+
+`BOFTConfig` allows you to control how OFT/BOFT is applied to the base model through the following parameters:
+
+- `boft_block_size`: the BOFT matrix block size across different layers, expressed in ``int``. Smaller block size results in sparser update matrices with fewer trainable paramters. **Note**, please choose it to be dividable to most layer ``in_features`` dimension, e.g., 4, 8, 16. Also, you can only specify either `boft_block_size` or `boft_block_num`, but not both simultaneously, because `boft_block_size` x `boft_block_num` = layer dimension.
```

Comment:
Hi @BenjaminBossan 
Thanks a lot for the detailed suggestions! 

1. Regarding the "compile" and "import" issue
We solve this by adding cuda file into pakage data(https://github.com/yfeng95/peft/blob/ca8f32c887841c01fe1a92424bae4a3a93db4276/setup.py#L48) and moving cuda loading inside boft layer (https://github.com/yfeng95/peft/blob/ca8f32c887841c01fe1a92424bae4a3a93db4276/src/peft/tuners/boft/layer.py#L176).  
So it will save the .cu data for pipe install, and automatically compile when and only when there's "import boft".

Reply:
Yes, we have removed the double-backticks in [pull/1326](https://github.com/huggingface/peft/pull/1326/files#diff-c292fd931d83a057e5efcd6a8e1e150c09d0bd6fd0cafa3653bfe9668ff74dceR65)